### PR TITLE
Fix sync function to support delete list correctly

### PR DESCRIPTION
**Pull request by pasin**
- Ensure to grant the user to the list channel when the list is deleted.
- Update database name from todos to todolite
- Set log to *

#54 and https://github.com/couchbase/sync_gateway/issues/1484#issuecomment-172695912

### CODE REVIEW

**Comment by pasin** [comment]
cc @tleyden @sethrosetter

File: sync-gateway-config.json [hunk 1]
```diff
@@ -12,38 +12,46 @@ function(doc, oldDoc) {
...
+      throw({forbidden : "items must have a list_id"});
     }
     channel("list-"+doc.list_id);
-  } else if (doc.type == "list") {
-    channel("list-"+doc._id);
-    if (!doc.owner) {
-      throw({forbidden : "list must have an owner"})
+  } else if (doc.type == "list" || (doc._deleted && oldDoc && oldDoc.type == "list")) {
+    // Make sure that only the owner of the list can update the list:
+    var owner = oldDoc ? oldDoc.owner : doc.owner;
```
Comment (adamcfraser) [inline sync-gateway-config.json]: It looks like this lets the oldDoc.owner change doc.owner to someone other than themselves?  Is that intentional?

Comment (pasin) [inline sync-gateway-config.json]: Will requireUser() at line 26 will reject the change if that happens? I'm not sure if I understand requireUser correctly or not.

Comment (adamcfraser) [inline sync-gateway-config.json]: If oldDoc != null, the requireUser at line 26 is requireUser(oldDoc.owner).

When oldDoc != null, you're no longer doing requireUser(doc.owner) anywhere (I don't think).

**Comment by pasin** [comment]
@adamcfraser I fixed the sync function per your comment. Please review when you have time. Thanks!

**Comment by adamcfraser** [comment]
@pasin A minor point - I might change the thrown error message when owner != doc.owner from "no permission to modify the doc." to "Cannot change owner for lists", so it's more transparent what the error actually is (for end users, and for people using this sync function as a reference).

**Comment by pasin** [comment]
Yes, I changed the message per your suggestion. Thanks!

**Comment by pasin** [comment]
I think we are good here so I'm merging this myself now.

**Pull request merged**